client.go

```diff
@@ -216,14 +216,11 @@ func (c *Client) exchangeDOH(ctx context.Context, m *Msg, a string) (r *Msg, rtt
 		return nil, 0, err
 	}
 
-	// TODO(tmthrgd): Allow the path to be customised?
-	u := &url.URL{
-		Scheme: "https",
-		Host:   a,
-		Path:   "/.well-known/dns-query",
-	}
-	if u.Port() == "443" {
-		u.Host = u.Hostname()
+	u, err := url.Parse(a)
+	if err != nil {
+		return nil, 0, err
+	} else if u.Scheme != "https" {
```

Comment:
should we just not care? garbage in; garbage out and having a http way of doing things might benefit some local testing?

Reply:
I think it’s best to be restrictive, the spec is specifically over HTTPS. Also anyone doing `Exchange(new(Msg), “http://dns.cloudflare/dns-query”)`, will see the request succeeded, but silently leak DNS requests before the redirect kicks in. Also note, there is no existing support for non-HTTPS in HEAD.